src/test/mir-opt/inline-closure.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-flags: -Z span_free_formats
+
+// Tests that MIR inliner can handle closure arguments. (#45894)
+
+fn main() {
+    println!("{}", foo(0, 14));
+}
+
+fn foo<T: Copy>(_t: T, q: i32) -> i32 {
+    let x = |_t, _q| _t;
+    x(q*2, q*3)
+}
+
+// END RUST SOURCE
+// START rustc.foo.Inline.after.mir
+// ...
+// bb0: {
+//     StorageLive(_3);
+//     _3 = [closure@NodeId(28)];
+//     StorageLive(_4);
+//     _4 = &_3;
+//     StorageLive(_5);
+//     StorageLive(_6);
+//     StorageLive(_7);
+//     _7 = _2;
+//     _6 = Mul(_7, const 2i32);
```

Comment:
Checked arithmetic strikes again! These are `CheckedMul` in debug build. Please change to a non-checked operator like `&` or maybe just write `x(q, q)`.

<details>

```
[01:06:23] ---- [mir-opt] mir-opt/inline-closure.rs stdout ----
[01:06:23] 	thread '[mir-opt] mir-opt/inline-closure.rs' panicked at 'Did not find expected line, error: Mismatch in lines
[01:06:23] Currnt block: None
[01:06:23] Expected Line: "        _8 = CheckedMul(_7, const 2i32); // scope 1 at /checkout/src/test/mir-opt/inline-closure.rs:21:7: 21:10"
[01:06:23] Actual Line: "    _6 = Mul(_7, const 2i32);"
[01:06:23] Expected:
[01:06:23] ... (elided)
[01:06:23] ... (elided)
[01:06:23] bb0: {
[01:06:23]     StorageLive(_3);
[01:06:23]     _3 = [closure@NodeId(28)];
[01:06:23]     StorageLive(_4);
[01:06:23]     _4 = &_3;
[01:06:23]     StorageLive(_5);
[01:06:23]     StorageLive(_6);
[01:06:23]     StorageLive(_7);
[01:06:23]     _7 = _2;
[01:06:23]     _6 = Mul(_7, const 2i32);
[01:06:23]     StorageDead(_7);
[01:06:23]     StorageLive(_8);
[01:06:23]     StorageLive(_9);
[01:06:23]     _9 = _2;
[01:06:23]     _8 = Mul(_9, const 3i32);
[01:06:23]     StorageDead(_9);
[01:06:23]     _5 = (_6, _8);
[01:06:23]     _0 = (_5.0: i32);
[01:06:23]     StorageDead(_5);
[01:06:23]     StorageDead(_8);
[01:06:23]     StorageDead(_6);
[01:06:23]     StorageDead(_4);
[01:06:23]     StorageDead(_3);
[01:06:23]     return;
[01:06:23] }
[01:06:23] ... (elided)
[01:06:23] Actual:
[01:06:23] fn foo(_1: T, _2: i32) -> i32 {
[01:06:23]     let mut _0: i32;
[01:06:23]     scope 1 {
[01:06:23]         let _3: [closure@NodeId(28)];
[01:06:23]         scope 3 {
[01:06:23]         }
[01:06:23]     }
[01:06:23]     scope 2 {
[01:06:23]     }
[01:06:23]     let mut _4: &[closure@NodeId(28)];
[01:06:23]     let mut _5: (i32, i32);
[01:06:23]     let mut _6: i32;
[01:06:23]     let mut _7: i32;
[01:06:23]     let mut _8: (i32, bool);
[01:06:23]     let mut _9: i32;
[01:06:23]     let mut _10: i32;
[01:06:23]     let mut _11: (i32, bool);
[01:06:23]     bb0: {                              
[01:06:23]         StorageLive(_3);
[01:06:23]         _3 = [closure@NodeId(28)];
[01:06:23]         StorageLive(_4);
[01:06:23]         _4 = &_3;
[01:06:23]         StorageLive(_5);
[01:06:23]         StorageLive(_6);
[01:06:23]         StorageLive(_7);
[01:06:23]         _7 = _2;
[01:06:23]         _8 = CheckedMul(_7, const 2i32);
[01:06:23]         assert(!(_8.1: bool), "attempt to multiply with overflow") -> bb1;
[01:06:23]     }
[01:06:23]     bb1: {                              
[01:06:23]         _6 = (_8.0: i32);
[01:06:23]         StorageDead(_7);
[01:06:23]         StorageLive(_9);
[01:06:23]         StorageLive(_10);
[01:06:23]         _10 = _2;
[01:06:23]         _11 = CheckedMul(_10, const 3i32);
[01:06:23]         assert(!(_11.1: bool), "attempt to multiply with overflow") -> bb2;
[01:06:23]     }
[01:06:23]     bb2: {                              
[01:06:23]         _9 = (_11.0: i32);
[01:06:23]         StorageDead(_10);
[01:06:23]         _5 = (_6, _9);
[01:06:23]         _0 = (_5.0: i32);
[01:06:23]         StorageDead(_5);
[01:06:23]         StorageDead(_9);
[01:06:23]         StorageDead(_6);
[01:06:23]         StorageDead(_4);
[01:06:23]         StorageDead(_3);
[01:06:23]         return;
[01:06:23]     }
[01:06:23] }', /checkout/src/tools/compiletest/src/runtest.rs:2337:12
[01:06:23] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[01:06:23] 
[01:06:23] 
[01:06:23] failures:
[01:06:23]     [mir-opt] mir-opt/inline-closure.rs
[01:06:23] 
[01:06:23] test result: [31mFAILED(B[m. 44 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
```

</details>